### PR TITLE
Split single sram address into two address spaces.

### PR DESCRIPTION
**Pull request by reclusejack**
We need to split the sram when there is only one sram address in the system(E20),
otherwise, we will suffer big drop in performance.

### CODE REVIEW

File: freedom-ldscript-generator.c++ [hunk 1]
```diff
@@ -348,18 +372,20 @@ static void write_linker_memory (fstream &os, bool scratchpad, uint32_t metal_en
...
           flash_offset = metal_entry;
           entry.mem_length -= metal_entry;
-	}
-	os << "\t" << entry.mem_alias <<  " (rxai!w)";
+        }
+        os << "\t" << entry.mem_alias <<  " (rxai!w)";
       } else if (entry.mem_alias.compare("ram") == 0) {
-	os << "\t" << entry.mem_alias <<  " (wxa!ri)";
+        os << "\t" << entry.mem_alias <<  " (wxa!ri)";
+        flash_offset = 0;
```
Comment (bsousi5) [inline freedom-ldscript-generator.c++]: This is not necessary.

Comment (reclusejack) [reply freedom-ldscript-generator.c++]: update the patch.

**Comment by bsousi5** [comment]
Sorry, I should have mention there were two of them.

**Comment by reclusejack** [comment]
The variable flash_offset will be modified to non-zero value after flash memory entry parsing.
But after split sram0, we will add sram1 to the end of the entry list, which is added after flash memory. So, we will get the wrong address base if we don't set flash_offset to 0.

**Comment by bsousi5** [comment]
Oh I see, I missed the fact that in this case it will be running from itim.
Got it. Look Thanks

**Pull request merged**